README.md

```diff
@@ -1,4 +1,39 @@
-forever-agent
-=============
+# forever-agent
 
-HTTP Agent that keeps socket connections alive between keep-alive requests. Formerly part of mikeal/request, now a standalone module.
+HTTP Agent that keeps socket connections alive between keep-alive requests.
+Formerly part of mikeal/request, now a standalone module.
+
+## Installation
+
+`forever-agent` can be installed with npm:
+```bash
+npm install forever-agent [--save]
+```
+
+## Usage
+
+The `ForeverAgent` and `ForeverAgent.SSL` are classes that extend `http.Agent`
+and `https.Agent` respectively.
+
+Many popular request libraries already support `forever-agent` out of the box,
+but global usage can be achieved with the following code snippet:
+
+```js
+var ForeverAgent = require("forever-agent");
+var http = require("http");
+var https = require("https");
+
+var options = {};
+http.globalAgent = new ForeverAgent(options);
```

Comment:
Does this actually works?
Have you checked that `http.globalAgent` can be replaced?
https://github.com/nodejs/node/issues/9057